packages/flutter_tools/lib/src/android/gradle.dart

```diff
@@ -93,29 +93,34 @@ Future<GradleProject> _readGradleProject() async {
   final FlutterProject flutterProject =  new FlutterProject(fs.currentDirectory);
   final String gradle = await _ensureGradle(flutterProject);
   await updateLocalProperties(project: flutterProject);
+  final Status status = logger.startProgress('Resolving dependencies...', expectSlowOperation: true);
+  GradleProject project;
   try {
-    final Status status = logger.startProgress('Resolving dependencies...', expectSlowOperation: true);
     final RunResult runResult = await runCheckedAsync(
       <String>[gradle, 'app:properties'],
       workingDirectory: flutterProject.android.directory.path,
       environment: _gradleEnv,
     );
     final String properties = runResult.stdout.trim();
-    final GradleProject project = new GradleProject.fromAppProperties(properties);
-    status.stop();
-    return project;
-  } catch (e) {
+    project = new GradleProject.fromAppProperties(properties);
+  } catch (exception) {
     if (getFlutterPluginVersion(flutterProject.android) == FlutterPluginVersion.managed) {
+      status.cancel();
       // Handle known exceptions. This will exit if handled.
-      handleKnownGradleExceptions(e);
+      handleKnownGradleExceptions(exception);
 
       // Print a general Gradle error and exit.
-      printError('* Error running Gradle:\n$e\n');
+      printError('* Error running Gradle:\n$exception\n');
       throwToolExit('Please review your Gradle project setup in the android/ folder.');
     }
+    // Fall back to the default
+    project = new GradleProject(
+      <String>['debug', 'profile', 'release'],
+      <String>[], flutterProject.android.gradleAppOutV1Directory,
+    );
   }
-  // Fall back to the default
-  return new GradleProject(<String>['debug', 'profile', 'release'], <String>[], flutterProject.android.gradleAppOutV1Directory);
+  status.stop();
```

Comment:
Wouldn't you want this in a `finally{}` block?

Reply:
If I put it in a finally block, it'll run after the `status.cancel` call in the `catch`. That's the one case I _don't_ want to run it in. All the other cases, as far as I can tell, fall through to this line without a `finally`.

Reply:
Yeah, I guess since you're catching all exceptions, that makes sense.  Could you catch more specific exceptions, or is it just too wide open what could be thrown? It always seems bad to me to catch everything: it masks a lot of runtime exceptions.